docs/developers.rst

```diff
@@ -394,6 +394,12 @@ Use ``mutable`` buffer instead of ``immutable var``:
    val buffer = scala.collection.mutable.ArrayBuffer.empty[Int]
    buffer += 1
    buffer += 2
+   
+**Additionally note that, for the best performance, use mutable inside of methods, but return an immutable**
+
+   val mutableList = scala.collection.mutable.MutableList(1,2)
+   mutableList += 3
+   mutableList.toList //returns scala.collection.immutable.List
```

Comment:
To avoid extra GC pressure due to creating the copy of mutable collection, one can introduce method which returns common type from `scala.collection`, for example:
```
def prep(n: Int): scala.collection.Seq[Int] = {
  val rnd = new scala.util.Random()
  val mutableList = scala.collection.mutable.MutableList[Int]()
  (1 to n).foreach { i =>
    val v = rnd.nextInt()
    mutableList += v
  }
  mutableList
}
```

Reply:
Addressed - Unless it becomes an explicit problem I'd rather leave the choice open - since this way is slightly more dangerous since it's still ultimately mutable.